packages/components/src/internal/components/security/APIWrapper.ts

```diff
@@ -53,6 +82,11 @@ export function getSecurityTestAPIWrapper(
     return {
         fetchContainers: mockFn(),
         fetchPolicy: mockFn(),
+        fetchGroups: mockFn(),
+        createGroup: mockFn(),
+        deleteGroup: mockFn(),
+        addGroupMembers: mockFn(),
+        removeGroupMembers: mockFn(),
```

Comment:
The better paradigm is to put in default implementations, Alan mentioned—I'm not sure exactly what that looks like for this case. Are there examples for me to check out?

Reply:
You probably don't need a default implementation for every method, but, you should have a default implementation for every one of these methods that gets automatically called when a component is mounted. If I had to guess fetchGroups is one of those. The rest are only going to be called when specific actions are taken (e.g. filling out a form then clicking a button), in those cases mockFn() is fine, and you can override as needed (in the test cases that trigger calls to the methods).

Reply:
Here is an example in ELN: https://github.com/LabKey/labbook/blob/develop/packages/eln/src/test/utils.tsx#L44

We have a default value for loadAuthors because we have a bunch of components that make a call to this on component mount. Before this implementation was just jest.fn(), and our test logs were riddled with errors, after this default implementation (and others) our test suite runs without any errors being emitted except those that are purposely emitted during tests.

Reply:
(In-progress for unit tests)